packages/react-slider/src/components/Slider/useSliderState.tsx

```diff
@@ -180,16 +175,9 @@ export const useSliderState = (state: Pick<SliderState, keyof SliderCommon | key
     }
   }, [currentValue, state.ref]);
 
-  useMount(() => {
-    // If the user passes an out of bounds controlled value, clamp and update their value onMount.
-    if (value !== undefined && (value < min || value > max) && onChange && onChangeCallback.current) {
```

Comment:
This isn't quite what I was thinking of--in the v8 Slider, we *do* clamp the value even if controlled, and reflect that in the UI and the `componentRef.value` (we just don't call `onChange` after clamping). However it's kind of a weird edge case where the proper handling in converged could be up for discussion. (I'm not sure what v0 does.)
https://github.com/microsoft/fluentui/blob/1f31459994791525f065b24dabc0c55baacf2f8f/packages/react/src/components/Slider/useSlider.ts#L109-L120

Reply:
It is definitely an interesting edge case and is up for discussion. After coming back to this, I'm leaning towards requiring the user to clamp their state as it might be to overbearing if the Slider does it for them. In the current implementation, the value internally is clamped but will not update the users controlled value through onChange. I believe the v8 and v0 do the same:
https://codesandbox.io/s/old-slider-clamping-cp9yg?file=/src/App.js